xds/src/main/java/io/grpc/xds/FallbackLb.java

```diff
@@ -0,0 +1,106 @@
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.Status;
+import io.grpc.internal.GrpcAttributes;
+import io.grpc.internal.ServiceConfigUtil.LbConfig;
+import io.grpc.util.ForwardingLoadBalancer;
+import io.grpc.util.GracefulSwitchLoadBalancer;
+import io.grpc.xds.XdsLoadBalancerProvider.XdsConfig;
+import java.util.List;
+import java.util.Map;
+
+/** Fallback load balancer. Handles fallback policy changes. */
+final class FallbackLb extends ForwardingLoadBalancer {
+
+  private final LoadBalancerRegistry lbRegistry;
+  private final GracefulSwitchLoadBalancer fallbackPolicyLb;
+
+  FallbackLb(Helper fallbackLbHelper) {
+    this(fallbackLbHelper, LoadBalancerRegistry.getDefaultRegistry());
+  }
+
+  @VisibleForTesting
+  FallbackLb(Helper fallbackLbHelper, LoadBalancerRegistry lbRegistry) {
+    this.lbRegistry = lbRegistry;
+    fallbackPolicyLb = new GracefulSwitchLoadBalancer(fallbackLbHelper);
+  }
+
+  @Override
+  protected LoadBalancer delegate() {
+    return fallbackPolicyLb;
+  }
+
+  @Override
+  public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+    // In the future, xdsConfig can be gotten directly by
+    // resolvedAddresses.getLoadBalancingPolicyConfig()
+    Attributes attributes = resolvedAddresses.getAttributes();
+    Map<String, ?> newRawLbConfig = checkNotNull(
+        attributes.get(ATTR_LOAD_BALANCING_CONFIG), "ATTR_LOAD_BALANCING_CONFIG not available");
+    ConfigOrError cfg =
+        XdsLoadBalancerProvider.parseLoadBalancingConfigPolicy(newRawLbConfig, lbRegistry);
+    if (cfg.getError() != null) {
+      throw cfg.getError().asRuntimeException();
+    }
+    XdsConfig xdsConfig = (XdsConfig) cfg.getConfig();
+
+    LbConfig fallbackPolicy = xdsConfig.fallbackPolicy;
+    String newFallbackPolicyName = fallbackPolicy.getPolicyName();
+    fallbackPolicyLb.switchTo(lbRegistry.getProvider(newFallbackPolicyName));
+
+    List<EquivalentAddressGroup> servers = resolvedAddresses.getAddresses();
+    // Some addresses in the list may be grpclb-v1 balancer addresses, so if the fallback policy
+    // does not support grpclb-v1 balancer addresses, then we need to exclude them from the list.
+    if (!newFallbackPolicyName.equals("grpclb") && !newFallbackPolicyName.equals(XDS_POLICY_NAME)) {
+      ImmutableList.Builder<EquivalentAddressGroup> backends = ImmutableList.builder();
+      for (EquivalentAddressGroup eag : resolvedAddresses.getAddresses()) {
+        if (eag.getAttributes().get(GrpcAttributes.ATTR_LB_ADDR_AUTHORITY) == null) {
+          backends.add(eag);
+        }
+      }
+      servers = backends.build();
+    }
+
+    // TODO(zhangkun83): FIXME(#5496): this is a temporary hack.
+    if (servers.isEmpty()
+        && !fallbackPolicyLb.canHandleEmptyAddressListFromNameResolution()) {
+      fallbackPolicyLb.handleNameResolutionError(Status.UNAVAILABLE.withDescription(
```

Comment:
You don't want to do this because this will trigger #5692. @ejona86, I think this again indicates the `canHandleEmptyAddressListFromNameResolution()` API doesn't work well. In #5803 we got away by saying `AutoConfiguredLoadBalancer` is not a `LoadBalancer`. We can do the same here. WDYT?

Reply:
The problem of #5692 is that name resolver will not refresh, but here it doesn't seem it needs to refresh. Xds policy receiving empty address list is a normal usecase, and fallback policy being round robin is also a normal usecase. As long as there is no resolution error at the top level xds policy, we don't need to refresh the resolver.

Reply:
If fallback is round-robin, but the addresses are empty, don't you need a refresh?

Reply:
If the xds policy does not explicitly provide fallback policy, then we use default fallback policy which is round-robin, and in this case the addresses being empty is normal . The fallback at startup timer is only 10 seconds, I think it does not make much sense to refresh resolver when the timer expires.

Reply:
I'd suggest this class's implementation of `canHandleEmptyAddressListFromNameResolution()` should delegate to `fallbackPolicyLb`, but `canHandleEmptyAddressListFromNameResolution()` is called before configuration is provided. That seems like a pretty serious failing. I agree that we should discuss alternatives with the other languages.

Reply:
As discussed offline. We can leave the behavior unchanged.